pytorch_lightning/loops/fit_loop.py

```diff
@@ -161,12 +161,12 @@ def done(self) -> bool:
                 )
         self.trainer.should_stop = should_stop
 
-        return stop_steps or should_stop or stop_epochs
+        return stop_steps or should_stop or stop_epochs or self.trainer.num_training_batches == 0
 
     @property
     def skip(self) -> bool:
         """Whether we should skip the training and immediately return from the call to :meth:`run`."""
-        return self.done or self.trainer.num_training_batches == 0
+        return self.done or self.trainer.limit_train_batches == 0
```

Comment:
@awaelchli @tchaton n00b question: why do we need `skip` at all vs directly calling `run` and having it exit?

Reply:
We believe it was cleaner to have a skip function if needed, so the code is slightly more organized. But it could be removed.